packages/components/src/text-control/types.ts

```diff
@@ -3,7 +3,22 @@
  */
 import type { BaseControlProps } from '../base-control/types';
 
-export type TextControlProps = Pick<
+export type TextControlPropsType =
```

Comment:
Can we obtain this union from the JSX namespace? Otherwise we'll need to manually maintain it and keep it in sync when new types are added.

Reply:
I believe it's intentional that we maintain this list manually, since not all `input` types are compatible with this component (e.g. range, radio, etc).

Reply:
Ah, I think I missed the fact that this list was already there before the change. I thought it was being added in this PR. You can disregard this note.